pyglmnet/utils.py

```diff
@@ -130,3 +138,41 @@ def set_log_level(verbose):
             raise ValueError('verbose must be of a valid type')
         verbose = logging_types[verbose]
     logger.setLevel(verbose)
+
+
+def _verbose_iterable(data):
+    """Wrap an iterable object with tqdm.
+
+    If tqdm is not available or if we did not set the appropriate
+    log level, then we fall back to the classical method.
+
+    Parameters
+    ----------
+    data: range, list \n
+        This will be data which will wrapped by tqdm (if available).
+
+    Returns
+    -------
+    wrapped_data: \n
+        Data object wrapped with tqdm.
+    """
+    wrapped_data = data
+    if logger.getEffectiveLevel() == logging.INFO:
+        if has_tqdm:
+            wrapped_data = tqdm(data, leave=False)
+    return wrapped_data
+
+
+def _tqdm_log(msg):
```

Comment:
@jasmainak I've added this new method here to switch easily from the standard logger to tqdm if we need to print some information within a tqdm loop. This makes sure that everything will look good inside the terminal/notebook.